### PR TITLE
34 bugfix pipe and consumercluster do not have required methods

### PR DESCRIPTION
**Pull request by samvanderzwan**

### CODE REVIEW

File: src/simulator_core/adapter/transforms/esdl_asset_mapper.py [hunk 1]
```diff
@@ -17,13 +19,20 @@ class EsdlAssetMapper(EsdlMapperAbstract):
     }
 
     def to_esdl(self, entity: AssetAbstract) -> Any:
-        """ Maps entity object to PyEsdl objects"""
+        """Maps entity object to PyEsdl objects."""
         raise NotImplementedError("EsdlAssetMapper.to_esdl()")
 
-    def to_entity(self, model: EsdlAssetObject) -> AssetAbstract:
+    def to_entity(self, model: EsdlAssetObject, **kwargs) -> AssetAbstract:
```
Comment (MichielTukker) [inline src/simulator_core/adapter/transforms/esdl_asset_mapper.py:25]: We should refactor this.   The required info (from_junction, to_junction, etc)  should already be available within the EsdlAssetObject

File: src/simulator_core/adapter/transforms/esdl_asset_mapper.py [hunk 2]
```diff
@@ -17,13 +19,20 @@ class EsdlAssetMapper(EsdlMapperAbstract):
...
+        """
         if not type(model.esdl_asset) in self.conversion_dict:
             raise NotImplementedError(str(model.esdl_asset) + ' not implemented in conversion')
-        return self.conversion_dict[type(model.esdl_asset)]()
+        return self.conversion_dict[type(model.esdl_asset)](model.esdl_asset.name,
+                                                            model.esdl_asset.id,
+                                                            kwargs['from_junction'],
+                                                            kwargs['to_junction'],
+                                                            kwargs['thermal_production'],
+                                                            kwargs['temperature_supply'])
```
Comment (MichielTukker) [inline src/simulator_core/adapter/transforms/esdl_asset_mapper.py:38]: If you're retrieving the required data from the kwargs param, there's no guarantee that these parameters are actually available (resulting in an error/exception from the wrong location).    These should be fixed parameters in the EsdlAssetObject

File: src/simulator_core/adapter/transforms/mappers.py [hunk 3]
```diff
@@ -44,7 +44,11 @@ def to_entity(self, model: EsdlObject) -> HeatNetwork:
         # create junctions
         # create assets
         # create connection between them
-        assets_list = [EsdlAssetMapper().to_entity(x)
+
+        assets_list = [EsdlAssetMapper().to_entity(x,
+                                                   from_junction=None,
+                                                   to_junction=None, thermal_production=0.0,
+                                                   temperature_supply=0.0)
```
Comment (MichielTukker) [inline src/simulator_core/adapter/transforms/mappers.py:51]: Template code?   why not just pass an empty list, or a list with a single object?

File: src/simulator_core/entities/assets/production_cluster.py [hunk 4]
```diff
@@ -111,15 +110,15 @@ def __init__(
...
-            temperature_supply=self.temperature_supply,
-            temperature_return=self.temperature_return,
-            pandapipes_net=self.pandapipes_net,
-        )
+#        self._controlled_mass_flow = heat_demand_and_temperature_to_mass_flow(
+#            thermal_demand=self.thermal_production_required,
+#            temperature_supply=self.temperature_supply,
+#            temperature_return=self.temperature_return,
+#            pandapipes_net=self.pandapipes_net,
+#        )
```
Comment (MichielTukker) [inline src/simulator_core/entities/assets/production_cluster.py:118]: Why is this removed?  please add a comment with reasoning

File: src/simulator_core/entities/assets/production_cluster.py [hunk 5]
```diff
@@ -111,15 +110,15 @@ def __init__(
...
         # Objects of the asset
         self._initialized = False
-        self._create()
+        #self._create()
```
Comment (MichielTukker) [inline src/simulator_core/entities/assets/production_cluster.py:121]: Add comment

File: src/simulator_core/simulation/mappers/mappers.py [hunk 6]
```diff
@@ -7,14 +7,11 @@
...
 class EsdlMapperAbstract(ABC, Generic[Entity, EsdlAsset]):
-    #Abstract class to be used for deriving mapper classes from esdl to our own classes.
+    # Abstract class to be used for deriving mapper classes from esdl to our own classes.
     @abstractmethod
     def to_esdl(self, entity: Entity) -> EsdlAsset:
         """Map an Entity to a EsdlAsset"""
 
     @abstractmethod
-    def to_entity(self, model: EsdlAsset) -> Entity:
+    def to_entity(self, model: EsdlAsset, **kwargs) -> Entity:
```
Comment (MichielTukker) [inline src/simulator_core/simulation/mappers/mappers.py:16]: Why add kwargs?  all data should be inside the EsdlAsset object.

**Review by MichielTukker — CHANGES_REQUESTED** [review]

File: src/simulator_core/adapter/transforms/esdl_asset_mapper.py (hunk 1 above)
Comment (samvanderzwan) [reply src/simulator_core/adapter/transforms/esdl_asset_mapper.py:25]: As discussed this has been refactored

File: src/simulator_core/adapter/transforms/esdl_asset_mapper.py (hunk 2 above)
Comment (samvanderzwan) [reply src/simulator_core/adapter/transforms/esdl_asset_mapper.py]: Refactor to only pass name and id

File: src/simulator_core/adapter/transforms/mappers.py (hunk 3 above)
Comment (samvanderzwan) [reply src/simulator_core/adapter/transforms/mappers.py]: Removed these arguments, since they where no longer required

File: src/simulator_core/entities/assets/production_cluster.py (hunk 4 above)
Comment (samvanderzwan) [reply src/simulator_core/entities/assets/production_cluster.py]: is now out of the code since it is moved to other function

File: src/simulator_core/entities/assets/production_cluster.py (hunk 5 above)
Comment (samvanderzwan) [reply src/simulator_core/entities/assets/production_cluster.py]: Create function is no longer called from constructor

File: src/simulator_core/simulation/mappers/mappers.py (hunk 6 above)
Comment (samvanderzwan) [reply src/simulator_core/simulation/mappers/mappers.py:16]: Has been removed with the refactor

**Review by MichielTukker — APPROVED** [review]
discussed
